src/subdomains/supporting/pricing/services/pricing.service.ts

```diff
@@ -127,10 +149,15 @@ export class PricingService {
     return rule;
   }
 
-  private async isPriceValid(from: string, to: string, price: number, rule?: Rule): Promise<boolean> {
+  private async isPriceValid(
+    from: string,
+    to: string,
+    price: number,
+    rule?: Rule,
+    rulePrice?: Price,
+  ): Promise<boolean> {
     if (!rule) return true;
 
-    const rulePrice = await this.getRulePrice(rule);
     const difference = Math.abs(rulePrice.price - price) / price;
```

Comment:
`rulePrice?.price` and maybe add `??` because rulePrice is optional and could be undefined